lib/util/stop-times.ts

```diff
@@ -0,0 +1,82 @@
+import { addDays, isBefore } from 'date-fns'
+
+import {
+  PatternDayStopTimes,
+  PatternStopTimes,
+  StopData,
+  Time
+} from '../components/util/types'
+
+import { getOrPutEntry } from './arrays'
+import {
+  getRouteIdForPattern,
+  getStopTimesByPattern,
+  patternComparator,
+  routeIsValid,
+  stopTimeComparator
+} from './viewer'
+
+function hasValidTimesAndRoute({
+  pattern,
+  route,
+  times
+}: PatternStopTimes): boolean {
+  return (
+    times &&
+    times.length !== 0 &&
+    routeIsValid(route, getRouteIdForPattern(pattern))
+  )
+}
+
+/** Filter any times according to time range set in config. */
+function isWithinDaysAhead(time: Time, now: Date, daysAhead: number) {
+  const departureTime = time.serviceDay + time.realtimeDeparture
+  return isBefore(departureTime, addDays(now, daysAhead))
+}
+
+/** Helper to sort and group stop times by pattern by service day */
+export function groupAndSortStopTimesByPatternByDay(
```

Comment:
This logic is super clean, but since the shape of the graphql stoptimes response is different it might have to be adapted in the future. I wonder if we shouldn't build it for the new API and then transform the old API into the new shape for now?

Reply:
This can come later